src/scripts/angry-tock.test.js

```diff
@@ -0,0 +1,319 @@
+  getApp,
+  utils: {
+    slack: { postMessage, sendDirectMessage },
+    tock: { get18FTockSlackUsers, get18FTockTruants },
+  },
+} = require("../utils/test");
+
+describe("Angry Tock", () => {
+  const load = async () =>
+    new Promise((resolve) => {
+      jest.isolateModules(() => {
+        const module = require("./angry-tock"); // eslint-disable-line global-require
+        resolve(module);
+      });
+    });
+
+  const app = getApp();
+  const scheduleJob = jest.spyOn(scheduler, "scheduleJob");
+  let clock;
+
+  beforeAll(() => {
+    clock = sinon.useFakeTimers();
+  });
+
+  beforeEach(() => {
+    clock.reset();
+    jest.resetAllMocks();
+
+    process.env.TOCK_API = "tock url";
+    process.env.TOCK_TOKEN = "tock token";
+    process.env.ANGRY_TOCK_REPORT_TO = "#channel,@user";
+    process.env.ANGRY_TOCK_TIMEZONE = "Asia/Tokyo";
+    process.env.ANGRY_TOCK_FIRST_TIME = "10:00";
+    process.env.ANGRY_TOCK_SECOND_TIME = "14:45";
+
+    get18FTockTruants.mockResolvedValue([
+      {
+        id: "tock1",
+        email: "user@one",
+        first_name: "User",
+        last_name: "One",
+        username: "employee1",
+      },
+      {
+        id: "tock2",
+        email: "user@two",
+        first_name: "User",
+        last_name: "Two",
+        username: "employee2",
+      },
+      {
+        id: "tock4",
+        email: "user@four",
+        first_name: "User",
+        last_name: "Four",
+        username: "employee4",
+      },
+    ]);
+
+    get18FTockSlackUsers.mockResolvedValue([
+      {
+        email: "user@one",
+        id: "tock1",
+        name: "User 1",
+        slack_id: "slack1",
+        user: "employee1",
+      },
+      {
+        email: "user@two",
+        id: "tock2",
+        name: "User 2",
+        slack_id: "slack2",
+        user: "employee2",
+      },
+      {
+        email: "user@three",
+        id: "tock3",
+        name: "User 3",
+        slack_id: "slack3",
+        user: "employee3",
+      },
+    ]);
+  });
+
+  afterAll(() => {
+    clock.restore();
+  });
+
+  it("issues a warning and does not schedule a shouting match if un/mis-configured", async () => {
+    process.env.TOCK_API = "";
+    process.env.TOCK_TOKEN = "";
+    let angryTock = await load();
+    angryTock(app);
+    expect(app.logger.warn).toHaveBeenCalled();
+    expect(scheduleJob).not.toHaveBeenCalled();
+
+    process.env.TOCK_API = "set";
+    process.env.TOCK_TOKEN = "";
+    angryTock = await load();
+    angryTock(app);
+    expect(app.logger.warn).toHaveBeenCalled();
+    expect(scheduleJob).not.toHaveBeenCalled();
+
+    process.env.TOCK_API = "";
+    process.env.TOCK_TOKEN = "set";
+    angryTock = await load();
+    angryTock(app);
+    expect(app.logger.warn).toHaveBeenCalled();
+    expect(scheduleJob).not.toHaveBeenCalled();
+  });
+
+  describe("schedules the next shouting match on startup", () => {
+    it("if it is shouty day and before first-shout time, schedules a first-shout", async () => {
```

Comment:
I like the way you're thinking. That's a better organization. I'm going to document these for #261 for post-migration refactor and cleanup, unless you think this should be done sooner rather than later!

Reply:
I can do a test cleanup pass after this gets merged in!